### ISSUE TITLE
Login Problems icloud v3 

### ISSUE THREAD

**Issue by kristinaminich** [gcobb321/icloud3_v3#168]
A few days ago I installed icloud3 v3 19.0 in Home Assistant and it worked like a charm.

But then I had to change my Apple-ID password and since then it is broken. I tried to delete it many times and reinstalled it and in the end I even installed v3 19.0.2-

The problem is the configuration. Every time I try to change the Apple-ID password and then enter the Apple verification code, it won't save the changes. I'm stuck in a loop.

1. I fill in the old apple ID with the new password on the configuration site.
2. I get the verification code on my mobile and fill it in.
3. The code gets accepted but it won't save and return to the main menu.
4. I can enter the same code again and again and will never be redirected to the main menu.
5. I restart icloud3 and the old password is filled back in and so icloud3 can't connect to the icloud server.

After several trials of installing/deleting/reinstalling I will always end with an unknown error occured when I fill in a new verification code.

What am I missing here? Is the old password saved anywhere where I can delete it? Every time I reinstall icloud3 the old password is automatically back there.

Thank you!

**Comment by gcobb321** [comment]
The Unknown Error Occurred message indicates there was some type of program error that iCloud3 was not handling properly. There will be error messages in the HA log file `config/home-assistant.log` or in the iCloud3 log file `config/icloud3-0.log`. 

I would like those messages to see what is going on and fix the error before I release it.  Email the `icloud3-0.log` and `icloud3-1.log` to geekstergary@gmail.com. Thanks. 

The Apple ID and password are stored in the username and password fields in the _core/.storage/icloud3/configuration_ file. Edit those fields and remove the text between the quotes. 

```
"data": {
        "tracking": {
            "username": "yourappleid@email.com",
            "password": "««R2NpdDMyMTc=»»"
            …
```
Becomes

```
"data": {
        "tracking": {
            "username": "",
            "password": ""
            …
```

Save the file and restart HA

**Comment by kristinaminich** [comment]
As I keep trying to find a solution, I always get the Message "An unknown error occured" whenever I try to fill in the verification code. I even changed my apple password again, but no chance.

**Comment by gcobb321** [comment]
There will be a detailed error message that points to the line in the program that caused the error. Another way of getting it is to go to Settings > System > Logs in HA. 

It will getting you a screen like this:
```
Home Assistant Core

Update of sensor.lillian_place is taking over 10 seconds
6:37:28 AM – (WARNING) helpers/entity.py - message first occurred at July 14, 2023 at 9:37:27 PM and shows up 25 times
Error handling request
12:14:44 AM – (ERROR) /usr/local/lib/python3.11/site-packages/aiohttp/web_protocol.py - message first occurred at July 14, 2023 at 11:57:47 PM and shows up 4 times
Error getting initial data for bedside_light.local: Read failed
July 14, 2023 at 7:50:37 PM – (WARNING) ESPHome
``` 

Look for the _Error Handling Request_  message from iCloud3. It will be near the top. Select it and you will see someone like there’s:
```
Logger: aiohttp.server
Source: /usr/local/lib/python3.11/site-packages/aiohttp/web_protocol.py:403 
First occurred: July 14, 2023 at 11:57:47 PM (4 occurrences) 
Last logged: 12:14:44 AM

Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_protocol.py", line 332, in data_received
    messages, upgraded, tail = self._request_parser.feed_data(data)
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "aiohttp/_http_parser.pyx", line 551, in aiohttp._http_parser.HttpParser.feed_data
aiohttp.http_exceptions.BadStatusLine: 400, message="Bad status line 'Invalid method encountered'" 
```

That will tell me what line in the program caused the error.  (That message is for the aiohttp.server program but it will look the same)

**Comment by kristinaminich** [comment]
Logger: aiohttp.server
Source: custom_components/icloud3/support/pyicloud_ic3.py:321
Integration: iCloud3 v3 ([documentation](https://gcobb321.github.io/icloud3_v3/#/), [issues](https://github.com/gcobb321/icloud3_v3/issues))
First occurred: 14:32:29 (1 occurrences)
Last logged: 14:32:29

Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_protocol.py", line 433, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 85, in security_filter_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 100, in forwarded_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 28, in request_context_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 80, in ban_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 236, in auth_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/view.py", line 148, in handle
    result = await handler(request, **request.match_info)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 242, in post
    return await super().post(request, flow_id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 72, in wrapper
    result = await method(view, request, data, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 110, in post
    result = await self._flow_mgr.async_configure(flow_id, data)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 297, in async_configure
    result = await self._async_handle_step(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 393, in _async_handle_step
    result: FlowResult = await getattr(flow, method)(user_input)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/icloud3/config_flow.py", line 1884, in async_step_reauth
    valid_code = await Gb.hass.async_add_executor_job(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/icloud3/support/pyicloud_ic3.py", line 987, in validate_2fa_code
    req = self.Session.post(f"{self.AUTH_ENDPOINT}/verify/trusteddevice/securitycode",
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 637, in post
    return self.request("POST", url, data=data, json=json, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/icloud3/support/pyicloud_ic3.py", line 232, in request
    self._raise_error(response.status_code, error_reason)
  File "/config/custom_components/icloud3/support/pyicloud_ic3.py", line 321, in _raise_error
    raise api_error
custom_components.icloud3.support.pyicloud_ic3.PyiCloudAPIResponseException:  (Status Code 401)

**Comment by kristinaminich** [comment]
I really appreciate your help! I hope I do understand what you want me to do because I'm not a computer engineer or something similar.

My whole Home Assistant Journey is a learning by doing.

**Comment by gcobb321** [comment]
Thanks. I think I have an idea of what it’s going on. I have tested changing the username/password when already logged into an account and that works fine. I have tested it when not logged into an account and the username/ password fields are empty. But I do not think I tested it when getting a username/password and trying to correct it. I’ll work in that. 

Clear the fields in the configuration file like I described above, restart HA and renter the username/password and that should work.
